### DISCUSSION TITLE
GitHub Copilot not working on mbp m1

### DISCUSSION THREAD

**Discussion by aiyogg** [community/community#6886] [Copilot Conversations]
Tips `command 'github.copilot.generate' not found ` and no icon on the status bar.

Some other info:
```
版本: 1.61.2
提交: 6cba118ac49a1b88332f312a8f67186f7f3c1643
日期: 2021-10-19T15:03:09.437Z
Electron: 13.5.1
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin arm64 20.6.0
```
VSCode and GitHub Copilot extensions are all the latest versions. I have tried reinstalling Copilot, VSCode, and relogin GitHub account in VSCode, but the problem is still there

**Comment by davecheney** [comment]
@aiyogg sorry you're having problems. What we believe is happening is if copilot cannot talk to github's api to get a personal access token to talk to the copilot servers it can fail to start up properly, leading to these errors. I think there is a new build coming soon that will give a more descriptive output if there is a network issue on copilot startup.

**Reply by aiyogg** [reply]
Thanks for the reply. Do I have some methods such as HTTP proxy to resolve the network issue?

**Reply by davecheney** [reply]
At the moment copilot doesn't support HTTP proxies. I hope we add that in the future, but I don't have a eta when or if that may happen.

**Reply by aiyogg** [reply]
I have another MBP with intel arch, when I log in to my GitHub account Copilot works well. There has the same VSCode setting in these two machines and uses the same network. Is there any other reason cause this issue?

**Reply by davecheney** [reply]
I cannot think of any reason why an intel macbook vs an m1 macbook would behave differently. From what we've seen in the issues reported so far, copilot can fail to startup when it cannot talk to github or copilot. Why that is remains something of a mystery so I hope that we'll have a new build out soon which will report to the user some more useful debugging information which you can relay to us if you're having problems.

**Reply by aiyogg** [reply]
But it's not any logs in the Output tab with Copilot. The tips likely `command 'github.copilot.toggleCopilot' not found` are the only message for me when I run command about with Copilot. I think need more info for users went some problems.

**Reply by davecheney** [reply]
I hope that we'll have a new build out soon which will report to the user some more useful debugging information which you can relay to us if you're having problems.

**Reply by aiyogg** [reply]
> [ERROR] [default] [2021-11-18T12:54:47.132Z] GitHub Copilot could not connect to server. Extension activation failed: "unable to verify the first certificate".   

It has some logs in the latest version.

It works well in my WebStorm.

**Reply by davecheney** [reply]
Hello, I'm sorry you are having difficulties. We've found that the copilot icon may not show up if copilot is not able to contact our servers. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Reply by aiyogg** [reply]
Thanks for your reply.  GFW is in my country, and I use a VPN to connect to the company's working network. I have tried to use HTTP Proxy on the VSCode setting but all are the same. Has any resolution to this problem?

**Reply by davecheney** [reply]
I’m sorry, at the moment copilot doesn’t support http proxies.
